packages/cli/src/environments.ee/source-control/source-control.service.ee.ts

```diff
@@ -481,6 +487,13 @@ export class SourceControlService {
 	async getStatus(user: User, options: SourceControlGetStatus) {
 		await this.sanityCheck();
 
+		const context = new SourceControlContext(user);
+
+		if (options.direction === 'pull' && !context.accessToAllProjects()) {
+			// A pull is only allowed by global admins or owners
+			return [];
```

Comment:
Why not throw an exception here, instead of returning empty array. Or even maybe making this check in the controller itself to throw a ForbiddenError maybe:
```ts
if (req.query.direction === 'pull' && !hasGlobalScope(req.user, 'sourceControl:pull')) {
  throw new ForbiddenError('You do not have permission to pull from source control');
}
```